v2/client.go

```diff
@@ -25,7 +26,7 @@ type Client struct {
 // NewClient creates new concordium grpc client.
 func NewClient(config Config) (_ *Client, err error) {
 	if config.TlsCredentials != nil {
-		conn, err := grpc.Dial(
+		conn, err := grpc.NewClient(
```

Comment:
What's the difference here? 🤔

Reply:
`Dial` is apparently deprecated and you should use NewClient instead. That's what the deprecation warning said :sweat_smile:. I don't really know what the difference is.

Reply:
According to the docs:

>One subtle difference between NewClient and Dial and DialContext is that the former uses "dns" as the default name resolver, while the latter use "passthrough" for backward compatibility. This distinction should not matter to most users, but could matter to legacy users that specify a custom dialer and expect it to receive the target string directly. 

Hopefully we are in that "most users" group :shrug: